### PR TITLE
Fix various test bugs seen when running on Windows

### PR DESCRIPTION
**Pull request by basil**
Fixes various minor bugs discovered in tests while running on Windows.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: test/src/test/java/hudson/PluginManagerTest.java
Comment (basil) [inline test/src/test/java/hudson/PluginManagerTest.java:197]: I unintentionally broke this test in commit 1b2ca8a709 by switching from `tasks.jpi` to `htmlpublisher.jpi` without updating the corresponding class name, but we never noticed because the return value is ignored. I fixed the return value problem by asserting that the value is not null, which would have exposed the original problem in the first place.

File: test/src/test/java/jenkins/security/stapler/Security914Test.java
Comment (basil) [inline test/src/test/java/jenkins/security/stapler/Security914Test.java:49]: We no longer bundle `credentials` as a detached plugin as of commit commit c6f061f26a, which this test wanted. Fortunately, there's a `credentials.hpi` in the source tree, so we can get the same effect with a `@WithPlugin` annotation.

**Review by timja — APPROVED** [review]

**Comment by basil** [comment]
This PR was tested successfully in [a passing Windows CI run](https://ci.jenkins.io/job/Core/job/jenkins/job/PR-6024/25/) in #6024.

**Review by jtnord — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
